script/check-arm-build-cmake

```diff
@@ -98,6 +98,8 @@ build_samr21()
 
 main()
 {
+    export CPPFLAGS="${CPPFLAGS:-} -DNDEBUG"
```

Comment:
Do we want to check on CI build only with `NDEBUG` set? From our experience, the majority of customers keep asserts in the deployed product anyway.

Reply:
This script is only used for CI check, and `script/cmake-build` doesn't have this flag. Adding test for the case without `NDEBUG` will double the check time. Since build without `NDEBUG` can easily be catched during daily work. I think it's fine to just check build with `NDEBUG`. Thoughts?

Reply:
Okay let's go in this path. Thanks.